### ISSUE TITLE
Links to PGS Catalog from GWAS search UI have disappeared

### ISSUE THREAD

**Issue by ljwh2** [EBISPOT/goci#435]
There used to be a button to link from the GWAS Catalog publication page and trait page to relevant pages in the PGS Catalog. I have screen shots of these from Sept 2020, but they are not there now. These should be reinstated. There should also have been one from the study page according to original specification, but I can't verify this. 

![Screenshot 2021-07-08 at 16.56.49.png](https://images.zenhubusercontent.com/5e57ecb5d0668b3a20e94b9e/b36b12bf-0403-47a1-8c34-98d6cbe847a2)![Screenshot 2021-07-08 at 16.52.34.png](https://images.zenhubusercontent.com/5e57ecb5d0668b3a20e94b9e/afd7ba78-000d-49bd-95f0-f0f11d19e593)

The screen shots are from
https://www.ebi.ac.uk/gwas/efotraits/EFO_0004340
https://www.ebi.ac.uk/gwas/publications/21743057

**Comment by ljwh2** [comment]
Original ticket to add these is here: https://github.com/EBISPOT/gwas-ui/issues/174

**Comment by ljwh2** [comment]
Info from Laurent: 

Looking at the logs, there are frequently calls from GWAS to our REST API endpoint to get the corresponding PGS ID and it looks like the requests worked.

Ah, this is likely because we changed the format of the /rest/publication/ endpoint (in February).
It used to be:
...
"associated_pgs_ids": [
    "PGS000010",
    "PGS000011"
]
...
but from February it is:
...
"associated_pgs_ids": {
    "development": [
        "PGS000011"
    ],
    "evaluation": [
        "PGS000010",
        "PGS000011"
    ]
}
...
Basically we split the list of associated PGS IDs in 2 different categories. This was mainly done to be consistent with the rest of the REST API. (edited)

**Comment by ala-ebi** [comment]
It seems this is a CORS problem. 
![image.png](https://images.zenhubusercontent.com/60190c8d2d567e0492e6fdd9/00c346f1-a83d-4337-b544-4cf271ed43a3)
This probably means PGS should add a the "Access-Control-Allow-Origin" response headers to their API responses.

**Comment by smlmbrt** [comment]
> It seems this is a CORS problem.
> ![image.png](https://camo.githubusercontent.com/1cd6ba90519846e27ca780f7d0617e9694d32b2a98eafbbfaaf93c78ee2befa8/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3630313930633864326435363765303439326536666464392f30306333343666312d613833642d343333372d623534342d346366323731656434336133)
> This probably means PGS should add a the "Access-Control-Allow-Origin" response headers to their API responses.

@ens-lgil could you look adding the proper CORS permissions on our end? Also it appears the GCST endpoint they need is also not working (e.g. https://www.pgscatalog.org/rest/gwas/get_score_ids/GCST004988), despite the matched landing page working (https://www.pgscatalog.org/gwas/GCST004988/).

**Comment by ens-lgil** [comment]
Hi, 
I fixed the regex and added a missing library for the CORS handling on the PGS Catalog side.
Please, @ala-ebi, can you check if it worls now ?
Thanks,
Laurent

**Comment by ala-ebi** [comment]
I just check and it works great. Thank you!

**Comment by ens-lgil** [comment]
Fantastic!
I also just fixed the issue with the REST endpoint: https://www.pgscatalog.org/rest/gwas/get_score_ids/GCST004988.
Thank you for reporting this issue as well.
